pkg/webhook/podprobemarker/validating/probe_create_update_handler.go

```diff
@@ -0,0 +1,235 @@
+			return allErrs
+		}
+		if k8sNativePodConditions.Has(probe.Name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("probes"), spec.Selector, fmt.Sprintf("probe name can't be %s", probe.Name)))
+			return allErrs
+		}
+		if uniqueProbe.Has(probe.Name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("probes"), spec.Selector, "probe name must be unique in PodProbeMarker."))
+			return allErrs
+		}
+		uniqueProbe.Insert(probe.Name)
+		allErrs = append(allErrs, validateProbe(&probe.Probe, fldPath.Child("probe"))...)
+		uniquePolicy := sets.NewString()
+		for _, policy := range probe.MarkerPolicy {
+			if uniquePolicy.Has(string(policy.State)) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("markerPolicy"), spec.Selector, "marker policy state must be unique."))
+				return allErrs
+			}
+			uniquePolicy.Insert(string(policy.State))
+			allErrs = append(allErrs, validateProbeMarkerPolicy(&policy, fldPath.Child("markerPolicy"))...)
+		}
+	}
+
+	return allErrs
+}
+
+func validateSelector(selector *metav1.LabelSelector, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, metavalidation.ValidateLabelSelector(selector, fldPath)...)
+	if len(selector.MatchLabels)+len(selector.MatchExpressions) == 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, selector, "empty selector is not valid for podProbeMarker."))
+	}
+	_, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("selector"), selector, ""))
+	}
+	return allErrs
+}
+
+func validateProbe(probe *appsv1alpha1.ContainerProbeSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if probe == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, probe, "probe can't be empty in podProbeMarker."))
+		return allErrs
+	}
+	allErrs = append(allErrs, validateHandler(&probe.Handler, fldPath)...)
+	allErrs = append(allErrs, validation.ValidateNonnegativeField(int64(probe.InitialDelaySeconds), fldPath.Child("initialDelaySeconds"))...)
+	allErrs = append(allErrs, validation.ValidateNonnegativeField(int64(probe.TimeoutSeconds), fldPath.Child("timeoutSeconds"))...)
+	allErrs = append(allErrs, validation.ValidateNonnegativeField(int64(probe.PeriodSeconds), fldPath.Child("periodSeconds"))...)
+	allErrs = append(allErrs, validation.ValidateNonnegativeField(int64(probe.SuccessThreshold), fldPath.Child("successThreshold"))...)
+	allErrs = append(allErrs, validation.ValidateNonnegativeField(int64(probe.FailureThreshold), fldPath.Child("failureThreshold"))...)
+	return allErrs
+}
+
+func validateHandler(handler *corev1.Handler, fldPath *field.Path) field.ErrorList {
+	numHandlers := 0
+	allErrors := field.ErrorList{}
+	if handler.Exec != nil {
+		numHandlers++
+		allErrors = append(allErrors, validateExecAction(handler.Exec, fldPath.Child("exec"))...)
+	}
+	if handler.HTTPGet != nil || handler.TCPSocket != nil {
```

Comment:
Should not check `handler.HTTPGet != nil || handler.TCPSocket != nil`, so that if there are both `HTTPGet` and `TCPSocket`, `numHandlers` will only be `1`